src/Tokenizer/Tokens.php

```diff
@@ -961,6 +961,10 @@ public function setCode($code)
             $this->registerFoundToken($token);
         }
 
+        if (defined('T_HH_ERROR') && $this->isTokenKindFound(T_HH_ERROR)) {
+            throw new \ParseError('Parsing error, encountered "T_HH_ERROR".');
```

Comment:
I don't find any way to make this consistent with PHP error message:
`"syntax error, unexpected 'asd' (T_STRING), expecting ...`.
under HHVM, you just got T_HH_ERROR for almost all tokens if parsing fails.